plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeAdvanceOptions.vue

```diff
@@ -0,0 +1,460 @@
+            />
+          </div>
+        </div>
+        <div class="row">
+          <div class="col s12">
+            <Field
+              uicontrol="text"
+              name="custom-campaign-keyword-query-param"
+              :model-value="customCampaignKeyword"
+              @update:model-value="customCampaignKeyword = $event; updateTrackingCode(this.site)"
+              :disabled="isLoading"
+              :title="translate('CoreAdminHome_JSTracking_CampaignKwdParam')"
+            />
+          </div>
+        </div>
+      </div>
+
+    </div>
+    <hr v-if="showBottomHR" class="hr-bottom">
+  </div>
+</template>
+<script lang="ts">
+import { defineComponent } from 'vue';
+import {
+  translate,
+  AjaxHelper,
+  SiteRef,
+  SelectOnFocus,
+  debounce,
+  Matomo,
+} from 'CoreHome';
+import { Field } from 'CorePluginsAdmin';
+
+interface CustomVar {
+  name: string;
+  value: string;
+}
+
+interface JsTrackingCodeGeneratorState {
+  showAdvanced: boolean;
+  site: SiteRef;
+  trackAllSubdomains: boolean;
+  isLoading: boolean;
+  siteUrls: Record<string, string[]>;
+  siteExcludedQueryParams: Record<string, string[]>,
+  siteExcludedReferrers: Record<string, string[]>,
+  crossDomain: boolean;
+  groupByDomain: boolean;
+  trackAllAliases: boolean;
+  trackNoScript: boolean;
+  trackCustomVars: boolean;
+  customVars: CustomVar[];
+  canAddMoreCustomVariables: boolean;
+  doNotTrack: boolean;
+  disableCookies: boolean;
+  useCustomCampaignParams: boolean;
+  customCampaignName: string;
+  customCampaignKeyword: string;
+  trackingCodeAbortController: AbortController | null;
+  isHighlighting: boolean;
+  consentManagerName: string;
+  consentManagerUrl: string;
+  consentManagerIsConnected: boolean;
+}
+
+interface GetJavascriptTagResponse {
+  value: string;
+}
+
+function getHostNameFromUrl(url: string) {
+  const urlObj = new URL(url);
+  return urlObj.hostname;
+}
+
+function getCustomVarArray(cvars: CustomVar[]) {
+  return cvars.map((cv) => [cv.name, cv.value]);
+}
+
+const piwikHost = window.location.host;
+const piwikPath = window.location.pathname.substring(0, window.location.pathname.lastIndexOf('/'));
+
+export default defineComponent({
+  props: {
+    defaultSite: {
+      type: Object,
+      required: true,
+    },
+    maxCustomVariables: Number,
+    serverSideDoNotTrackEnabled: Boolean,
+    showBottomHR: Boolean,
+  },
+  data(): JsTrackingCodeGeneratorState {
+    return {
+      showAdvanced: false,
+      site: this.defaultSite as SiteRef,
+      trackAllSubdomains: false,
+      isLoading: false,
+      siteUrls: {},
+      siteExcludedQueryParams: {},
+      siteExcludedReferrers: {},
+      crossDomain: false,
+      groupByDomain: false,
+      trackAllAliases: false,
+      trackNoScript: false,
+      trackCustomVars: false,
+      customVars: [],
+      canAddMoreCustomVariables: !!this.maxCustomVariables && this.maxCustomVariables > 0,
```

Comment:
This should suffice, shouldn't it? Unless we can compare the actual number with the max number and disable based on that, it will always be false when 0 and always true when non-zero, so either of the conditions on its own is fine.
```suggestion
      canAddMoreCustomVariables: this.maxCustomVariables > 0,
```

Reply:
@michalkleiner It was added at the time of angular to vue migration, there could be some edge case which I might not be aware off, so didn't refactor this part